src/duckdb.mjs

```diff
@@ -1,361 +1,228 @@
-        const stmt = await conn.prepare(query);
-        result = stmt.query(...params);
-      } else {
-        result = await conn.query(query);
+  async query(query, params) {
+    const result = await this.queryStream(query, params);
+    const results = [];
+    for await (const rows of result.readRows()) {
+      for (const row of rows) {
+        results.push(row);
       }
-      return result;
-    }
-
-    // The `.queryStream` function will supercede this for SQL and Table cells
-    // Keeping this for backwards compatibility
-    async sql(strings, ...args) {
-      // expected to be used like db.sql`select * from table where foo = ${param}`
-
-      const results = await this.query(strings.join("?"), args);
-
-      // return rows as a JavaScript array of objects for now
-      let rows = results.toArray().map(Object.fromEntries);
-      rows.columns = results.schema.fields.map((d) => d.name);
-      return rows;
-    }
-
-    async table(query, params, opts) {
-      const result = await this.query(query, params);
-      return Inputs.table(result, {
-        layout: "auto",
-        ...(opts || {})
-      });
-    }
-
-    // get the client after the query ran
-    async client(query, params) {
-      await this.query(query, params);
-      return this;
-    }
-
-    // query a single row
-    async queryRow(query, params) {
-      const key = `Query ${this._counter++}: ${query}`;
-      const conn = await this.connection();
-      // use send as we can stop iterating after we get the first batch
-      const result = await conn.send(query, params);
-      const batch = (await result.next()).value;
-      return batch && batch.get(0);
-    }
-
-    async explain(query, params) {
-      const row = await this.queryRow(`EXPLAIN ${query}`, params);
-      return element("pre", {
-        className: "observablehq--inspect"
-      }, [
-        text(row["explain_value"])
-      ]);
-    }
-
-    // Describe the database (no arg) or a table
-    async describe(object) {
-      const result = await (object === undefined ?
-        this.query(`SHOW TABLES`) :
-        this.query(`DESCRIBE ${object}`));
-      return Inputs.table(result);
     }
+    results.schema = result.schema;
+    return results;
+  }
 
-    // Summarize a query result
-    async summarize(query) {
-      const result = await this.query(`SUMMARIZE ${query}`);
-      return Inputs.table(result);
-    }
+  async queryRow(query, params) {
+    const results = await this.query(query, params);
+    return results.length ? results[0] : null;
+  }
 
-    async insertJSON(name, buffer, options) {
-      const db = await this.db();
-      await db.registerFileBuffer(name, new Uint8Array(buffer));
-      const conn = await db.connect();
-      await conn.insertJSONFromPath(name, {
-        name,
-        schema: "main",
-        ...options
-      });
-      await conn.close();
+  async sql(strings, ...args) {
+    return await this.query(strings.join("?"), args);
```

Comment:
Same comment as above (should we just use `queryStream`?)

Reply:
No, then we’d need to duplicate the rest of _db_.query. _db_.queryStream doesn’t return a promise to an array of results; it returns a [query stream response](https://observablehq.com/@observablehq/database-client-specification#%C2%A73.2).

Reply:
Got it, thanks for the clarification!